libs/libc/string/lib_memcpy.c

```diff
@@ -34,7 +34,9 @@
  * Name: memcpy
  ****************************************************************************/
 
-#ifndef CONFIG_LIBC_ARCH_MEMCPY
+#if !defined(CONFIG_LIBC_ARCH_MEMCPY) || \
+    (defined(CONFIG_LIBC_ARCH_USER_ONLY) && defined(__KERNEL__)) || \
+    (defined(CONFIG_LIBC_ARCH_KERNEL_ONLY) && !defined(__KERNEL__))
```

Comment:
should we add the similar guard to the optimization version under libc/machince?

Reply:
I added a guard for `nuttx/libs/libc/machine/<arch>` code (basically, the opposite of this) to avoid both the software version and the arch's being compiled. Could you please take a look at it?

Reply:
Done!